### ISSUE TITLE
None always returned from `AutoML.deploy()`

### ISSUE THREAD

**Issue by barnybug** [aws/sagemaker-python-sdk#1207]
### System Information
- **Framework**: AutoML
- **Python Version**: 3.7

### Describe the problem
`sagemaker.automl.automl.AutoML.deploy()` currently can't return a predictor. This is because you can't provide the predictor_cls parameter to the `PipelineModel()` constructed inside `_deploy_inference_pipeline`.

The docs for `sagemaker.automl.automl.AutoML.deploy()` say:
```
Returns:
            callable[string, sagemaker.session.Session]: Invocation of
            ``self.predictor_cls`` on the created endpoint name.
```

### Minimal repro / logs

Roughly:
```
from sagemaker.automl import automl
ml = automl.AutoML(...)
ml.fit(...)
candidate = ml.best_candidate(...)
predictor = ml.deploy(
    initial_instance_count=1,
    instance_type='ml.t2.medium',
    candidate=candidate,
)
predictor # always None
```

**Comment by laurenyu** [comment]
thanks for the detailed bug report! I've submitted a PR that should address this - once the change is released, you can specify `predictor_cls` through `AutoML.deploy()`

**Comment by laurenyu** [comment]
fix has been released in [v1.50.2](https://github.com/aws/sagemaker-python-sdk/releases/tag/v1.50.2)
